### ISSUE TITLE
[Bug] panic: interface conversion: interface {} is *schema.Set, not []string

### ISSUE THREAD

**Issue by SHolzhauer** [elastic/terraform-provider-elasticstack#336]
**Describe the bug**
When deploying spaces and adding the `disabled_features` option an error occurs which fails the apply step.

**To Reproduce**
1. TF configuration used
```tf
resource "elasticstack_kibana_space" "team_space" {
  name              = "test"
  space_id          = "test"
  disabled_features = ["ingestManager", "enterpriseSearch", "siem", "fleet", "fleetv2", "generalCases", "stackAlerts", "monitoring", "dev_tools", "osquery", "securitySolutionCases"]
}
```
3. TF operations to execute to get the error `terraform apply`
4. See the error in the output:
```
Stack trace from the terraform-provider-elasticstack_v0.6.0 plugin:
panic: interface conversion: interface {} is *schema.Set, not []string
goroutine 96 [running]:
github.com/elastic/terraform-provider-elasticstack/internal/kibana.resourceSpaceUpsert({0x1493d68, 0xc0003ed080}, 0xc0004ef080, {0x1246340, 0xc00020a6e0})
	github.com/elastic/terraform-provider-elasticstack/internal/kibana/space.go:94 +0x659
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0xc00038c460, {0x1493da0, 0xc0003e6000}, 0xd?, {0x1246340, 0xc00020a6e0})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/resource.go:741 +0x12e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc00038c460, {0x1493da0, 0xc0003e6000}, 0xc0005315f0, 0xc0004eef00, {0x1246340, 0xc00020a6e0})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/resource.go:847 +0x845
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000011278, {0x1493da0?, 0xc0003d1ec0?}, 0xc0005b9860)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/grpc_provider.go:1021 +0xe8d
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.muxServer.ApplyResourceChange({0xc000374ae0, 0xc000374b40, {0xc0003fc220, 0x1, 0x1}, {0x0, 0x0, 0x0}, {0x0, 0x0, ...}, ...}, ...)
	github.com/hashicorp/terraform-plugin-mux@v0.9.0/tf5muxserver/mux_server_ApplyResourceChange.go:27 +0x102
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc000348000, {0x1493da0?, 0xc0003d0390?}, 0xc0002879d0)
	github.com/hashicorp/terraform-plugin-go@v0.15.0/tfprotov5/tf5server/server.go:818 +0x574
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x128b840?, 0xc000348000}, {0x1493da0, 0xc0003d0390}, 0xc000287960, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.15.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:419 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0000ec000, {0x1497e20, 0xc00009a820}, 0xc00076d7a0, 0xc0003feed0, 0x1dadc00, 0x0)
	google.golang.org/grpc@v1.54.0/server.go:1345 +0xdf3
google.golang.org/grpc.(*Server).handleStream(0xc0000ec000, {0x1497e20, 0xc00009a820}, 0xc00076d7a0, 0x0)
	google.golang.org/grpc@v1.54.0/server.go:1722 +0xa36
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.54.0/server.go:966 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.54.0/server.go:964 +0x28a
Error: The terraform-provider-elasticstack_v0.6.0 plugin crashed!
This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
2023-05-26T07:34:07.544Z [DEBUG] provider: plugin exited
```

**Expected behavior**
I expect the space to be created/updated and the features in the list to be disabled.

**Debug output**
```
...
2023-05-26T07:34:07.275Z [TRACE] checkPlannedChange: Verifying that actual change (action Update) matches planned change (action Update)
[0m[1mmodule.spaces["uxresearch"].elasticstack_kibana_space.team_space: Modifying... [id=DBs10QuzTjei-3CmXM4zMA/uxresearch][0m[0m
2023-05-26T07:34:07.275Z [INFO]  Starting apply for module.spaces["uxresearch"].elasticstack_kibana_space.team_space
2023-05-26T07:34:07.276Z [DEBUG] module.spaces["uxresearch"].elasticstack_kibana_space.team_space: applying the planned Update change
2023-05-26T07:34:07.276Z [TRACE] GRPCProvider: ApplyResourceChange
[0m[1mmodule.spaces["wps"].elasticstack_kibana_space.team_space: Modifying... [id=DBs10QuzTjei-3CmXM4zMA/wps][0m[0m
2023-05-26T07:34:07.276Z [INFO]  Starting apply for module.spaces["wps"].elasticstack_kibana_space.team_space
2023-05-26T07:34:07.277Z [DEBUG] module.spaces["wps"].elasticstack_kibana_space.team_space: applying the planned Update change
2023-05-26T07:34:07.277Z [TRACE] checkPlannedChange: Verifying that actual change (action Update) matches planned change (action Update)
2023-05-26T07:34:07.277Z [TRACE] GRPCProvider: ApplyResourceChange
2023-05-26T07:34:07.277Z [TRACE] provider.terraform-provider-elasticstack_v0.6.0: Received request: @module=sdk.proto tf_provider_addr=registry.terraform.io/elastic/elasticstack tf_req_id=c1318465-f99b-0d56-7c98-1b2ac3f4e511 tf_resource_type=elasticstack_kibana_space tf_rpc=ApplyResourceChange @caller=github.com/hashicorp/terraform-plugin-go@v0.15.0/tfprotov5/tf5server/server.go:805 tf_proto_version=5.3 timestamp=2023-05-26T07:34:07.277Z
2023-05-26T07:34:07.278Z [TRACE] provider.terraform-provider-elasticstack_v0.6.0: Sending request downstream: @caller=github.com/hashicorp/terraform-plugin-go@v0.15.0/tfprotov5/internal/tf5serverlogging/downstream_request.go:17 @module=sdk.proto tf_rpc=ApplyResourceChange tf_proto_version=5.3 tf_provider_addr=registry.terraform.io/elastic/elasticstack tf_req_id=c1318465-f99b-0d56-7c98-1b2ac3f4e511 tf_resource_type=elasticstack_kibana_space timestamp=2023-05-26T07:34:07.277Z
2023-05-26T07:34:07.278Z [TRACE] provider.terraform-provider-elasticstack_v0.6.0: calling downstream server: tf_mux_provider=*schema.GRPCProviderServer tf_rpc=ApplyResourceChange @caller=github.com/hashicorp/terraform-plugin-mux@v0.9.0/internal/logging/mux.go:16 @module=sdk.mux timestamp=2023-05-26T07:34:07.277Z
2023-05-26T07:34:07.278Z [TRACE] provider.terraform-provider-elasticstack_v0.6.0: Calling downstream: @caller=github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/resource.go:846 tf_mux_provider=*schema.GRPCProviderServer tf_req_id=c1318465-f99b-0d56-7c98-1b2ac3f4e511 @module=sdk.helper_schema tf_provider_addr=registry.terraform.io/elastic/elasticstack tf_resource_type=elasticstack_kibana_space tf_rpc=ApplyResourceChange timestamp=2023-05-26T07:34:07.278Z
2023-05-26T07:34:07.278Z [TRACE] provider.terraform-provider-elasticstack_v0.6.0: Served request: tf_resource_type=elasticstack_kibana_space @caller=runtime/panic.go:884 @module=sdk.proto tf_proto_version=5.3 tf_provider_addr=registry.terraform.io/elastic/elasticstack tf_req_id=c1318465-f99b-0d56-7c98-1b2ac3f4e511 tf_rpc=ApplyResourceChange timestamp=2023-05-26T07:34:07.278Z
2023-05-26T07:34:07.278Z [DEBUG] module.spaces["fdfi"].elasticstack_kibana_space.team_space: applying the planned Update change
[0m[1mmodule.spaces["op"].elasticstack_kibana_space.team_space: Modifying... [id=DBs10QuzTjei-3CmXM4zMA/op][0m[0m
2023-05-26T07:34:07.279Z [INFO]  Starting apply for module.spaces["op"].elasticstack_kibana_space.team_space
2023-05-26T07:34:07.279Z [TRACE] GRPCProvider: ApplyResourceChange
2023-05-26T07:34:07.279Z [DEBUG] module.spaces["op"].elasticstack_kibana_space.team_space: applying the planned Update change
2023-05-26T07:34:07.280Z [TRACE] GRPCProvider: ApplyResourceChange
2023-05-26T07:34:07.280Z [DEBUG] provider.terraform-provider-elasticstack_v0.6.0: panic: interface conversion: interface {} is *schema.Set, not []string
2023-05-26T07:34:07.280Z [DEBUG] provider.terraform-provider-elasticstack_v0.6.0: 
2023-05-26T07:34:07.280Z [DEBUG] provider.terraform-provider-elasticstack_v0.6.0: goroutine 96 [running]:
2023-05-26T07:34:07.280Z [DEBUG] provider.terraform-provider-elasticstack_v0.6.0: github.com/elastic/terraform-provider-elasticstack/internal/kibana.resourceSpaceUpsert({0x1493d68, 0xc0003ed080}, 0xc0004ef080, {0x1246340, 0xc00020a6e0})
2023-05-26T07:34:07.280Z [DEBUG] provider.terraform-provider-elasticstack_v0.6.0: 	github.com/elastic/terraform-provider-elasticstack/internal/kibana/space.go:94 +0x659
2023-05-26T07:34:07.280Z [DEBUG] provider.terraform-provider-elasticstack_v0.6.0: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0xc00038c460, {0x1493da0, 0xc0003e6000}, 0xd?, {0x1246340, 0xc00020a6e0})
2023-05-26T07:34:07.280Z [DEBUG] provider.terraform-provider-elasticstack_v0.6.0: 	github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/resource.go:741 +0x12e
2023-05-26T07:34:07.280Z [DEBUG] provider.terraform-provider-elasticstack_v0.6.0: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc00038c460, {0x1493da0, 0xc0003e6000}, 0xc0005315f0, 0xc0004eef00, {0x1246340, 0xc00020a6e0})
2023-05-26T07:34:07.280Z [DEBUG] provider.terraform-provider-elasticstack_v0.6.0: 	github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/resource.go:847 +0x845
2023-05-26T07:34:07.280Z [DEBUG] provider.terraform-provider-elasticstack_v0.6.0: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000011278, {0x1493da0?, 0xc0003d1ec0?}, 0xc0005b9860)
2023-05-26T07:34:07.280Z [DEBUG] provider.terraform-provider-elasticstack_v0.6.0: 	github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/grpc_provider.go:1021 +0xe8d
2023-05-26T07:34:07.281Z [DEBUG] provider.terraform-provider-elasticstack_v0.6.0: github.com/hashicorp/terraform-plugin-mux/tf5muxserver.muxServer.ApplyResourceChange({0xc000374ae0, 0xc000374b40, {0xc0003fc220, 0x1, 0x1}, {0x0, 0x0, 0x0}, {0x0, 0x0, ...}, ...}, ...)
2023-05-26T07:34:07.281Z [DEBUG] provider.terraform-provider-elasticstack_v0.6.0: 	github.com/hashicorp/terraform-plugin-mux@v0.9.0/tf5muxserver/mux_server_ApplyResourceChange.go:27 +0x102
2023-05-26T07:34:07.281Z [DEBUG] provider.terraform-provider-elasticstack_v0.6.0: github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc000348000, {0x1493da0?, 0xc0003d0390?}, 0xc0002879d0)
2023-05-26T07:34:07.281Z [DEBUG] provider.terraform-provider-elasticstack_v0.6.0: 	github.com/hashicorp/terraform-plugin-go@v0.15.0/tfprotov5/tf5server/server.go:818 +0x574
2023-05-26T07:34:07.281Z [DEBUG] provider.terraform-provider-elasticstack_v0.6.0: github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x128b840?, 0xc000348000}, {0x1493da0, 0xc0003d0390}, 0xc000287960, 0x0)
2023-05-26T07:34:07.281Z [DEBUG] provider.terraform-provider-elasticstack_v0.6.0: 	github.com/hashicorp/terraform-plugin-go@v0.15.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:419 +0x170
2023-05-26T07:34:07.281Z [DEBUG] provider.terraform-provider-elasticstack_v0.6.0: google.golang.org/grpc.(*Server).processUnaryRPC(0xc0000ec000, {0x1497e20, 0xc00009a820}, 0xc00076d7a0, 0xc0003feed0, 0x1dadc00, 0x0)
2023-05-26T07:34:07.281Z [DEBUG] provider.terraform-provider-elasticstack_v0.6.0: 	google.golang.org/grpc@v1.54.0/server.go:1345 +0xdf3
2023-05-26T07:34:07.281Z [DEBUG] provider.terraform-provider-elasticstack_v0.6.0: google.golang.org/grpc.(*Server).handleStream(0xc0000ec000, {0x1497e20, 0xc00009a820}, 0xc00076d7a0, 0x0)
2023-05-26T07:34:07.281Z [DEBUG] provider.terraform-provider-elasticstack_v0.6.0: 	google.golang.org/grpc@v1.54.0/server.go:1722 +0xa36
2023-05-26T07:34:07.281Z [DEBUG] provider.terraform-provider-elasticstack_v0.6.0: google.golang.org/grpc.(*Server).serveStreams.func1.2()
2023-05-26T07:34:07.281Z [DEBUG] provider.terraform-provider-elasticstack_v0.6.0: 	google.golang.org/grpc@v1.54.0/server.go:966 +0x98
2023-05-26T07:34:07.281Z [DEBUG] provider.terraform-provider-elasticstack_v0.6.0: created by google.golang.org/grpc.(*Server).serveStreams.func1
2023-05-26T07:34:07.281Z [DEBUG] provider.terraform-provider-elasticstack_v0.6.0: 	google.golang.org/grpc@v1.54.0/server.go:964 +0x28a
2023-05-26T07:34:07.286Z [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: read unix @->/tmp/plugin127084363: read: connection reset by peer"
2023-05-26T07:34:07.286Z [ERROR] plugin.(*GRPCProvider).ValidateResourceConfig: error="rpc error: code = Unavailable desc = error reading from server: read unix @->/tmp/plugin127084363: read: connection reset by peer"
2023-05-26T07:34:07.286Z [ERROR] vertex "module.spaces[\"futureadminsogeti\"].elasticstack_kibana_space.team_space" error: Plugin did not respond
2023-05-26T07:34:07.286Z [TRACE] vertex "module.spaces[\"futureadminsogeti\"].elasticstack_kibana_space.team_space": visit complete, with errors
2023-05-26T07:34:07.287Z [TRACE] readDiff: Read Update change from plan for module.spaces["alacarte"].elasticstack_kibana_space.team_space
2023-05-26T07:34:07.287Z [TRACE] readResourceInstanceState: reading state for module.spaces["alacarte"].elasticstack_kibana_space.team_space
2023-05-26T07:34:07.286Z [ERROR] plugin.(*GRPCProvider).ValidateResourceConfig: error="rpc error: code = Unavailable desc = error reading from server: read unix @->/tmp/plugin127084363: read: connection reset by peer"
2023-05-26T07:34:07.286Z [ERROR] plugin.(*GRPCProvider).ApplyResourceChange: error="rpc error: code = Unavailable desc = error reading from server: read unix @->/tmp/plugin127084363: read: connection reset by peer"
2023-05-26T07:34:07.286Z [ERROR] plugin.(*GRPCProvider).ApplyResourceChange: error="rpc error: code = Unavailable desc = error reading from server: read unix @->/tmp/plugin127084363: read: connection reset by peer"
2023-05-26T07:34:07.286Z [ERROR] plugin.(*GRPCProvider).ApplyResourceChange: error="rpc error: code = Unavailable desc = error reading from server: read unix @->/tmp/plugin127084363: read: connection reset by peer"
2023-05-26T07:34:07.286Z [ERROR] plugin.(*GRPCProvider).PlanResourceChange: error="rpc error: code = Unavailable desc = error reading from server: read unix @->/tmp/plugin127084363: read: connection reset by peer"
2023-05-26T07:34:07.286Z [ERROR] plugin.(*GRPCProvider).PlanResourceChange: error="rpc error: code = Unavailable desc = error reading from server: read unix @->/tmp/plugin127084363: read: connection reset by peer"
2023-05-26T07:34:07.286Z [ERROR] plugin.(*GRPCProvider).ApplyResourceChange: error="rpc error: code = Unavailable desc = error reading from server: read unix @->/tmp/plugin127084363: read: connection reset by peer"
2023-05-26T07:34:07.286Z [ERROR] plugin.(*GRPCProvider).ApplyResourceChange: error="rpc error: code = Unavailable desc = error reading from server: read unix @->/tmp/plugin127084363: read: connection reset by peer"
2023-05-26T07:34:07.286Z [ERROR] plugin.(*GRPCProvider).ApplyResourceChange: error="rpc error: code = Unavailable desc = error reading from server: read unix @->/tmp/plugin127084363: read: connection reset by peer"
2023-05-26T07:34:07.286Z [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/elastic/elasticstack/0.6.0/linux_amd64/terraform-provider-elasticstack_v0.6.0 pid=112 error="exit status 2"
2023-05-26T07:34:07.287Z [TRACE] upgradeResourceState: schema version of module.spaces["alacarte"].elasticstack_kibana_space.team_space is still 0; calling provider "elasticstack" for any other minor fixups
2023-05-26T07:34:07.288Z [TRACE] GRPCProvider: UpgradeResourceState
2023-05-26T07:34:07.288Z [ERROR] plugin.(*GRPCProvider).UpgradeResourceState: error="rpc error: code = Canceled desc = context canceled"
...
```

**Versions (please complete the following information):**
 - OS: Linux
 - Terraform Version1.3.9
 - Provider version 0.6.0
 - Elasticsearch Version 8.6.1

**Comment by RobsonSutton** [comment]
Hey @SHolzhauer 👋 

I've just raised the above PR which should contain a fix to hopefully address the issues you have been seeing with the disabled_features field 👍

**Comment by tobio** [comment]
Fixed in 0.6.1
